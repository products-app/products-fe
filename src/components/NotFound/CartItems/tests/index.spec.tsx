import NotFoundCartItems from '../index'
import { render, screen } from '@testing-library/react'

describe('Not Found Cart Items component', () => {
  it('should render correctly', () => {
    const { container } = render(<NotFoundCartItems />)
    expect(container).toMatchSnapshot()
  })

  it('should contain the not found cart items message', () => {
    render(<NotFoundCartItems />)
    expect(
      screen.getByText('Adicione algo em seu carrinho...'),
    ).toBeInTheDocument()
    expect(() => screen.getByText('no exist')).toThrow()
  })

  it('should contain the not found cart items image', () => {
    render(<NotFoundCartItems />)
    const emptyCartImage = screen.getByRole('img')
    expect(emptyCartImage).toHaveAttribute('src')
    expect(emptyCartImage).toHaveAttribute('alt')
  })
})
