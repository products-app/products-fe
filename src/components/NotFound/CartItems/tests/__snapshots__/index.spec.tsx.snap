// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found Cart Items component should render correctly 1`] = `
<div>
  <div
    class="c-dzUtkR"
  >
    <img
      alt="Imagem de carrinho vazio"
      data-nimg="1"
      decoding="async"
      height="0"
      loading="lazy"
      sizes="100vw"
      src="/_next/image?url=%2Fempty-cart.png&w=3840&q=75"
      srcset="/_next/image?url=%2Fempty-cart.png&w=640&q=75 640w, /_next/image?url=%2Fempty-cart.png&w=750&q=75 750w, /_next/image?url=%2Fempty-cart.png&w=828&q=75 828w, /_next/image?url=%2Fempty-cart.png&w=1080&q=75 1080w, /_next/image?url=%2Fempty-cart.png&w=1200&q=75 1200w, /_next/image?url=%2Fempty-cart.png&w=1920&q=75 1920w, /_next/image?url=%2Fempty-cart.png&w=2048&q=75 2048w, /_next/image?url=%2Fempty-cart.png&w=3840&q=75 3840w"
      style="color: transparent;"
      width="0"
    />
    <p
      class="c-dgMvMW c-igsGei c-dgMvMW-kWyXas-size-md"
    >
      Adicione algo em seu carrinho...
    </p>
  </div>
</div>
`;
